### PR TITLE
bsc#1179007 and bsc#1179420, Bind address should not write to configuration file when using unicast and support enable secauth

### PR DESCRIPTION
**Pull request by nick-wang**
bsc#1179007, Do not write the bind address to configuration file when unicast.
Fix issue of configure ttl of the second ring.
bsc#1179420, fix secauth and support crypto_cipher/crypto_hash

### CODE REVIEW

**Comment by nick-wang** [comment]
This issue also only happened on ncurses (not GTK)... so also need yast team to help on checking the libraries.

**Comment by nick-wang** [comment]
above commit is for bsc#1179420, fix secauth and support crypto_cipher/crypto_hash

File: src/include/cluster/dialogs.rb [hunk 1]
```diff
@@ -511,10 +513,11 @@ def CommunicationLayout
 
       riface = CheckBoxFrame(
         Id(:enable2),
-        Opt(:notify),
+        Opt(:noAutoEnable, :notify),
```
Comment (jreidinger) [inline src/include/cluster/dialogs.rb:516]: what is this opt? I think it does not exists. What it should do?

File: src/include/cluster/dialogs.rb [hunk 2]
```diff
@@ -1078,6 +1073,19 @@ def SecurityDialog
...
+            HBox(
+              HSpacing(20),
+              Left(ComboBox(
+                Id(:crypto_hash), Opt(:hstretch, :notify), _("Crypto Hash:"),
+                ["sha1", "sha256", "sha384", "sha512", "md5"]
+              )),
+              HSpacing(5),
+              Left(ComboBox(
+                Id(:crypto_cipher), Opt(:hstretch, :notify), _("Crypto Cipher:"),
+                ["aes256", "aes192", "aes128", "3des"]
```
Comment (jreidinger) [inline src/include/cluster/dialogs.rb:1085]: just note that values won't be translated. I am not sure if e.g. chinese has special characters for it or not.

File: src/include/cluster/dialogs.rb (hunk 1 above)
Comment (nick-wang) [reply src/include/cluster/dialogs.rb:516]: It is caused by a wired issue of https://bugzilla.suse.com/show_bug.cgi?id=1179007

To short:
Explicit disable the checkbox doesn't work in ncurses UI, but works for others... And only this box.  Petr from L3 team give a possible analyze and a workaround of ":noAutoEnable" to make the parent box **disabled** by default. And it works in the test...

File: src/include/cluster/dialogs.rb (hunk 2 above)
Comment (nick-wang) [reply src/include/cluster/dialogs.rb:1085]: Do you mean "aes256", "3des" or "md5"? No, we also using the English word to represent it in Chinese world.

**Review by jreidinger — APPROVED** [review]

**Pull request merged**